### PR TITLE
UI updates part 4

### LINKED ISSUE #1106: Theme button is unstable
**Describe the bug**
Meng reports one problem is that the "theme" bar is always jumping up and down in the middle of the screen which is annoying.

Please make an additional change, which is not to allow the theme button when a user is at any of these links with 
https://staging.optimallearning.org/experiment/
in the URL. The experiments should not have this teacher. Ideally, the theme could be set in the tdf setspec in such cases if it were used.

### PR DESCRIPTION
**Pull request by JRustyHaner**
fixes #1106 and implements more UI improvements

@imrryr The tdf selection box has been implemented, so this is a major UI change.

image

### CODE REVIEW

File: mofacts/client/views/home/profile.html [hunk 1]
```diff
@@ -31,88 +31,124 @@
...
-                                                <br>
-                                                <p style="display:inline-block" title="Audio Output available for this module"><span class="fa fa-headphones"></span></p>
-                                            {{/if}}
-                                            <p style="display:inline-block">&nbsp;&nbsp;&nbsp;</p>
-                                            {{#if this.audioInputEnabled}}
-                                                <p style="display:inline-block" title="Speech Input available for this module"><i class="fa fa-microphone"></i></p>
-                                            {{/if}}
-                                        </button>
-                                        </div>
+                        <div class="col-12 text-center" id="practiceTDFSearchResultsContainer">
```
Comment (MegaGeese) [inline mofacts/client/views/home/profile.html:47]: This container covers the whole screen and prevents the user from being able to launch any TDFs

**Review by MegaGeese — CHANGES_REQUESTED** [review]

File: mofacts/client/views/home/profile.html (hunk 1 above)
Comment (JRustyHaner) [reply mofacts/client/views/home/profile.html:47]: doi, opacity instead of visibility.

Comment (JRustyHaner) [reply mofacts/client/views/home/profile.html:47]: @MegaGeese it's safe now

File: mofacts/client/index.html [hunk 2]
```diff
@@ -45,7 +46,9 @@ <h5 class="modal-title">Error Reporting</h5>
             </div>
           </div>
           <div class="row">
-            <div class="col-md-12">
+            <div class="col-md-12">He is fine
```
Comment (MegaGeese) [inline mofacts/client/index.html:49]: Im glad he's ok, but idk if we should tell the world.

File: mofacts/client/views/home/profile.js [hunk 3]
```diff
@@ -89,22 +93,54 @@ Template.profile.helpers({
...
+  'keyup #practiceTDFSearch': function(event, instance) {
     const search = event.target.value;
     const enabledTdfs = instance.enabledTdfs.get();
-    const filteredTdfs = enabledTdfs.filter((tdf) => {
+    filteredTdfs = enabledTdfs.filter((tdf) => {
       return tdf.tdfs.tutor.setspec.lessonname.toLowerCase().includes(search.toLowerCase());
     });
+    //also search tags
+    filteredTdfs = filteredTdfs.concat(enabledTdfs.filter((tdf) => {
+      return tdf.tdfs.tutor.setspec.tags && tdf.tdfs.tutor.setspec.tags.toLowerCase().includes(search.toLowerCase());
```
Comment (MegaGeese) [inline mofacts/client/views/home/profile.js:133]: Searching for a TDF breaks here. 
`profile.js:133 Uncaught TypeError: tdf.tdfs.tutor.setspec.tags.toLowerCase is not a function`

File: mofacts/client/views/home/profile.js [hunk 4]
```diff
@@ -89,22 +93,54 @@ Template.profile.helpers({
 // Template Events
 
 Template.profile.events({
-
+  //Enable TDF
+  'click .enableTdf': function(event, instance) {
```
Comment (MegaGeese) [inline mofacts/client/views/home/profile.js:97]: this doesnt do anything. `enableTdf` doesn't reference anything.

**Review by MegaGeese — CHANGES_REQUESTED** [review]
Files in the "new_templates" folder should be moved to more appropriate locations.

File: mofacts/client/views/home/profile.js (hunk 4 above)
Comment (JRustyHaner) [reply mofacts/client/views/home/profile.js:97]: > Files in the "new_templates" folder should be moved to more appropriate locations.

No. I need them there to keep track of what I've done.  That was purposeful. Changing the UI is a pretty lengthy process, so I need to keep track of them. This doesn't effect the build whatsoever.

> this doesn't do anything. `enableTdf` doesn't reference anything.

It wil l.This doesn't effect the build whatsoever.  The point of these incremental updates was to test them. There will be times in the near future where I write code before it's UI is implemented. 

> Searching for a TDF breaks here. `profile.js:133 Uncaught TypeError: tdf.tdfs.tutor.setspec.tags.toLowerCase is not a function`

Cannot reproduce.  

image

Comment (JRustyHaner) [reply mofacts/client/views/home/profile.js:97]: Actually, that last one I can. I don't have access to most of the TDFs, so I can't test tdfs with tags that are arrays. Let me fix that one right quick

**Pull request merged**